Source/CarthageKit/Dependency.swift

```diff
@@ -2,16 +2,39 @@ import Foundation
 import Result
 import Tentacle
 
+/// Protocol which supports URL + potential custom description
+public protocol URLConvertible: CustomStringConvertible {
+	var url: URL { get }
+}
+
 /// Uniquely identifies a project that can be used as a dependency.
 public enum Dependency {
+	/// Uniquely identifies a Binary Spec's resolved URL and its description
+	public struct BinaryURL: URLConvertible {
+		/// A Resolved URL
+		public let url: URL
+
+		/// An Optional custom description
+		public let overloadedDescription: String?
+
+		public var description: String {
+			return overloadedDescription ?? url.description
+		}
+
+		init(url: URL, overloadedDescription: String? = nil) {
+			self.url = url
+			self.overloadedDescription = overloadedDescription
+		}
+	}
+
 	/// A repository hosted on GitHub.com or GitHub Enterprise.
 	case gitHub(Server, Repository)
 
 	/// An arbitrary Git repository.
 	case git(GitURL)
 
 	/// A binary-only framework
-	case binary(URL)
+	case binary(BinaryURL)
```

Comment:
I was attempting to leverage generic protocol URLConvertible, but ran into:
https://www.bignerdranch.com/blog/why-associated-type-requirements-become-generic-constraints/

In this case, I feel the separate protocol does not provide enough benefit to exist on it's own. So I feel I can toss the protocol in favor with the publicly available BinaryURL